packages/web3-eth/src/web3_eth.ts

```diff
@@ -200,7 +200,15 @@ export class Web3Eth extends Web3Context<Web3EthExecutionAPI, RegisteredSubscrip
 		transactionHash: Bytes,
 		returnFormat: ReturnFormat = DEFAULT_RETURN_FORMAT as ReturnFormat,
 	) {
-		return rpcMethodsWrappers.getTransaction(this, transactionHash, returnFormat);
+		const response = await rpcMethodsWrappers.getTransaction(
+			this,
+			transactionHash,
+			returnFormat,
+		);
+
+		if (!response) throw new Error('Transaction not found!');
```

Comment:
I think we should use consistent approach for known errors in 4.x, like https://github.com/ChainSafe/web3.js/blob/e6a9aea25d3ce808f9a1cab4a26164366b68d3f4/packages/web3-errors/src/errors/transaction_errors.ts#L122   so could you add some thing like `TransactionNotFound` new error.